optuna/trial.py

```diff
@@ -351,6 +351,10 @@ def report(self, value, step=None):
         If step is set to :obj:`None`, the value is stored as a final value of the trial.
         Otherwise, it is saved as an intermediate value.
 
+        Note that the reported value is converted to ``float`` type by applying ``float()``
+        function internally. Thus, it accepts all float-like types (e.g., ``numpy.float32``).
+        If the conversion is failed, a ``TypeError`` exception is raised.
```

Comment:
Maybe this is my personal preference, `TypeError exception` is a precise expression, but it seems to be redundant.
```suggestion
        If the conversion is failed, a ``TypeError`` is raised.
```

Reply:
I agree with you. Thank you.